### ISSUE TITLE
cli: Add option to use a table formatter like the one provided by `psql`

### ISSUE THREAD

**Issue by rmloveland** [cockroachdb/cockroach#30851]
> **Is your feature request related to a problem? Please describe.**

I really miss the old style "bordered table" output in the CockroachDB CLI.  The old format could be pasted directly into [the built-in table editor in Org mode](https://orgmode.org/manual/Built_002din-table-editor.html#Built_002din-table-editor), for example.

> **Describe the solution you'd like**

I would like an option to output a table with borders similar to what `psql` has, e.g.,

```
defaultdb=> \pset border 2 
Border style is 2.
defaultdb=> EXPLAIN SELECT * FROM kv WHERE v BETWEEN 4 AND 5;
+------+-------+-------------+
| tree | field | description |
+------+-------+-------------+
| scan |       |             |
|      | table | kv@v        |
|      | spans | /4-/6       |
+------+-------+-------------+
(3 rows)
```

> **Describe alternatives you've considered**

A couple of alternatives I have considered:

- creating an Emacs macro to "borderize" the SQL output for those cases when I want to save it in an Org document
- using `psql` with `\pset border 2` more often (but I haven't learned how to connect it to secure clusters, or if it's possible)

**Comment by jlitzingerdev** [comment]
I've been looking at https://github.com/cockroachdb/cockroach/issues/30856, in which this issue is referenced.  Would the goal here be full parity with psql's border options, or only `border=2`?

For background, I've experimented a bit and the `border=2` option is straightforward, however, `border=0` doesn't *look* doable with the current vendored version of tablewriter.  More recent versions do have a SetNoWhiteSpace() option which doesn't quite emulate `border=0` but moves in the right direction.

**Comment by rmloveland** [comment]
> I've been looking at #30856, in which this issue is referenced. Would the goal here be full parity with psql's border options, or only border=2?

My strong *personal* preference is for `border=2`.

However, it looks like "the world has moved on" in our doc set to the borderless look, so it's probably much more useful to work on not generating trailing whitespace as discussed in the issue you linked.

However, if you felt like hacking on *both* of those things, nobody could stop you. :-)

Achieving full parity with psql's border options is not a goal AFAICT.

**Comment by knz** [comment]
Note that we need to fix #30856 regardless of what happens with this issue (users who wish to not use borders will still want to avoid trailing whitespaces).

I've sent a patch upstream for #30856.
